### DISCUSSION TITLE
it need to disconnect Pair topic after disconnecting session? IOS SDK

### DISCUSSION THREAD

**Discussion by xdozorx** [WalletConnect/walletconnect-monorepo#3628] [Developer Support] [answered]
When I disconnect a session, pair continues to be in storage.  `Pair.instance.getPairings()`

```
Task {
    try await Sign.instance.disconnect(topic: session.topic)
}
```

And when app is starting at launch, wc connect subscribe to topic, which contain in Pair. But session is finished.

It need to remove Pair, when session is disconnected? Because we have extra subscribe on wc, which is not good. 
In example of usage SDK, Pairs are not disconnecting

**Comment by mshahjahan-mufin** [comment, marked as answer]
@xdozorx You will need to disconnect session and pairing both.
